src/theory/fp/theory_fp_rewriter.cpp

```diff
@@ -787,6 +788,118 @@ namespace constantFold {
     }
   }
 
+  RewriteResponse componentFlag(TNode node, bool)
+  {
+    Kind k = node.getKind();
+
+    Assert((k == kind::FLOATINGPOINT_COMPONENT_NAN)
+           || (k == kind::FLOATINGPOINT_COMPONENT_INF)
+           || (k == kind::FLOATINGPOINT_COMPONENT_ZERO)
+           || (k == kind::FLOATINGPOINT_COMPONENT_SIGN));
+
+    FloatingPoint arg0(node[0].getConst<FloatingPoint>());
+
+    bool result;
+    switch (k)
+    {
+#ifdef CVC4_USE_SYMFPU
+      case kind::FLOATINGPOINT_COMPONENT_NAN:
+        result = arg0.getLiteral().nan;
+        break;
+      case kind::FLOATINGPOINT_COMPONENT_INF:
+        result = arg0.getLiteral().inf;
+        break;
+      case kind::FLOATINGPOINT_COMPONENT_ZERO:
+        result = arg0.getLiteral().zero;
+        break;
+      case kind::FLOATINGPOINT_COMPONENT_SIGN:
+        result = arg0.getLiteral().sign;
+        break;
+#endif
+      default: Unreachable("Unknown kind used in componentFlag"); break;
+    }
+
+    BitVector res(1U, (result) ? 1U : 0U);
+
+    return RewriteResponse(REWRITE_DONE,
+                           NodeManager::currentNM()->mkConst(res));
+  }
+
+  RewriteResponse componentExponent(TNode node, bool)
+  {
+    Assert(node.getKind() == kind::FLOATINGPOINT_COMPONENT_EXPONENT);
+
+    FloatingPoint arg0(node[0].getConst<FloatingPoint>());
+
+    // \todo Add a proper interface for this sort of thing to FloatingPoint
```

Comment:
What about this one?

Reply:
Ooops!  These ... it's hard to figure out what to do.  They are guarded by CVC4_USE_SYMFPU and the todo is ... it's not clear if they should be in with "avoid horrible death if floating-point things are used without CVC4_USE_SYMFPU" and I'm not sure how possible that is or whether it would just wind up duplicating effort.

I'll open an issue.

Reply:
https://github.com/CVC4/CVC4/issues/1915 Done.